### DISCUSSION TITLE
[Proposal] Spilt uno repository

### DISCUSSION THREAD

**Discussion by workgroupengineering** [unoplatform/uno#5422] [Ideas]
Hi,
when trying to debug the Uno source code, the compilation is long. : D.

What I would like to suggest is to split it into multiple repositories.
A possible subdivision could be this:

- Uno.Builds: contains all the tools necessary for the compilation and the various tools (SourceGenerator, T4Generator, etc ...)
- Uno.Tests: Contains common classes to run all tests
- Uno.UI.Runtime: the runtimes
- Uno.UI: the main repository
- Uno.VisualStudio.Exensibility: contains all projects concerning integration with VS.es: RemoteControl *, SolutionTemplate *, Analyzzer, etc.

Dependency on the repository would be handled through the nuget

This solution would also reduce the duration of the Azure pipeline because it wouldn't have to recompile everything every time.

**Comment by jeromelaban** [comment]
Thanks for the suggestion. At this point, creating multiple repositories is not a path we want to choose to handle dependencies, as most of those are internal artifacts used to create the build. The compilation time is not caused by the different parts of the solution, but rather the number of platforms built.

If you want to have a faster compilation time, [make sure to use solution filters](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/building-uno-ui.md#building-unoui-for-a-single-target-platform).
